docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,395 @@
+of new token programs, we need a set of conformance tests based on the current
+SPL token test suite.
+
+A new token program must include all of the same instructions as SPL token,
+but the "unchecked" variants should return errors. Concretely, this means the
+program must implement:
+
+* `InitializeMint`
+* `InitializeHolding` (previously `InitializeAccount`)
+* `Revoke`
+* `SetAuthority`
+* `CloseHolding` (previously `CloseAccount`)
+* `FreezeHolding` (previously `FreezeAccount`)
+* `ThawHolding` (previously `ThawAccount`)
+* `TransferChecked`
+* `ApproveChecked`
+* `MintToChecked`
+* `BurnChecked`
+* `InitializeHoldingWithOwnerAddress` (previously `InitializeAccount2`)
+
+And the program should throw errors for:
+
+* `Transfer`
+* `Approve`
+* `MintTo`
+* `Burn`
+* `InitializeMultisig` (use a more general multisig)
+
+New instructions required:
+
+* `CreateHolding`: only performs `Allocate` and `Assign` to self,
+useful for creating a holding when you don't know the program that you're
+interacting with. See the Associated Token Account section for how to use this.
+* `TransferAllChecked`: transfers everything from the source holding to the
+destination holding. Useful for closing a interest-bearing token holding since the
+balance is constantly changing.
+
+There are also new read-only instructions, which write data to a provided account
+buffer:
+
+* `NormalizeHolding`: write holding data in SPL token format
+* `NormalizeMint`: write mint data in SPL token format
+
+See the Runtime section for more information on how these are used.
+
+Programs and wallets that wish to support i-tokens must update the instructions
+to use all `Checked` variants and use the new read-only instructions to fetch
+information.
+
+The SPL token library provides wrappers compatible with any conforming program.
+For the base SPL token program, the wrapper simply deserializes the account data
+and returns the relevant data. For other programs, it calls the appropriate
+read-only instruction using an ephemeral account input.
+
+For interest-bearing tokens, after `NormalizeHolding` and deserialization
+into a `spl_token::Holding`, the `amount` field will be in terms of tokens, and
+not shares, so that UIs can properly display amounts in terms of tokens.
+
+### Token Program Registry
+
+We need the [token-list](https://github.com/solana-labs/token-list) to include
+vetted SPL token-conforming programs and group known mints by their program id.
+
+The token list will publish a file of known program ids at
+[token-list releases](https://github.com/solana-labs/token-list/releases/latest),
+to be used by RPC and programs.
+
+The entries in the token-list are signed using a managed key. As wallets or
+Ledgers are asked to send transactions using a particular token program, they can
+cross-check the mint or program information using a signature check on the 
+token-list data.
+
+TODO consider adding Rust and C versions of the registry for on-chain programs,
```

Comment:
I think it will be sufficient to add support to our _client_ libraries for fetching, caching and querying the token-list. God help us if that ever includes a C implementation :sweat_smile:

Reply:
right, i'll clarify the client library bit.  and good point about C integration :laughing: